### PR TITLE
feat(material): add prototype dialog component w/ demo.

### PR DESCRIPTION
**Pull request by jelbourn**
In its current state, this component is primarily meant to demonstrate how to use the `DynamicComponentLoader` to implement a dialog service. This will eventually evolve into the actual material design dialog.

Feel free to add better name suggestions.

### CODE REVIEW

File: modules/angular2_material/src/components/dialog/dialog.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+import {bind, Injector} from 'angular2/di';
+import {ObservableWrapper, Promise, PromiseWrapper} from 'angular2/src/facade/async';
+import {isPresent, Type} from 'angular2/src/facade/lang';
+import {DOM} from 'angular2/src/dom/dom_adapter';
+import {MouseEvent, KeyboardEvent} from 'angular2/src/facade/browser';
+import {KEY_ESC} from 'angular2_material/src/core/constants'
+
+
+// TODO(jelbourn): Opener of dialog can control where it is rendered.
+// TODO(jelbourn): Pressing `ESC` can close the dialog. Blocked on #1539
```
Comment (vsavkin) [inline modules/angular2_material/src/components/dialog/dialog.js]: This has been fixed.

Comment (jelbourn) [inline modules/angular2_material/src/components/dialog/dialog.js]: I'll add the behavior in the next PR.

File: modules/angular2_material/src/components/dialog/dialog.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    this.componentLoader = loader;
+  }
+
+  /**
+   * Opens a modal dialog.
+   * @param type The component to open.
+   * @param elementRef The logical location into which the component will be opened.
+   * @returns Promise for a reference to the dialog.
+   */
+  open(type: Type, elementRef: ElementRef, options: MdDialogConfig = null): Promise<MdDialogRef> {
```
Comment (vsavkin) [inline modules/angular2_material/src/components/dialog/dialog.js]: Can you add a todo here to replace MdDialogConfig with some interface? As a JS/TS developer I would like to use an object literal as a config and not create MgDialogConfig

**Comment by vsavkin** [comment]
Could you change it to use hostProperties and hostListeners where possible instead of direct dom access? The fix for dynamic components landed in master yesterday.

**Comment by vsavkin** [comment]
Also, I think we should test our material components. Maybe add a few e2e tests?

**Comment by jelbourn** [comment]
Interfaces don't work for the js.dev build right now AFAIK. 

I'm working on adding tests now.

**Comment by jelbourn** [comment]
(updated for the many changes that happened over the past few days)

I'm going to leave the DOM config until #1402 since the properties are static. Unit testing is somewhat blocked on #1609 and #1610, not sure if I want to have e2e for these now since they're still in a rough form. @vsavkin thoughts?

**Comment by vsavkin** [comment]
I'd add a few smoke e2e tests, but it is up to you.

**Pull request merged**